### PR TITLE
resource/cloudflare_email_routing_catch_all: switch to a dedicated scheme

### PR DESCRIPTION
**Pull request by yan12125**
Compared to a normal email routing rule, the catch-all rule:

* Does not have a priority
* For matchers, it accepts only type = "all", and there are no "field" and "value" properties
* For actions, it accepts type = "drop" besides "forward" and "worker"

Ref: https://api.cloudflare.com/#email-routing-routing-rules-update-routing-rule
Ref: https://api.cloudflare.com/#email-routing-routing-rules-update-catch-all-rule

Fixes https://github.com/cloudflare/terraform-provider-cloudflare/issues/1937

### CODE REVIEW

**Comment by yan12125** [comment]
As a side note, an API response for the catch-all rule contains `priority`. For example,
```json
{
 "result": {
  "tag": "xxx",
  "name": "Catch-all rule",
  "matchers": [
   {
    "type": "all"
   }
  ],
  "actions": [
   {
    "type": "drop"
   }
  ],
  "enabled": true,
  "priority": 2147483647
 },
 "success": true,
 "errors": [],
 "messages": []
}
```
I'm not sure if `priority` should be added as a computed field in the new scheme or not.

**Comment by jacobbednarz** [comment]
> I'm not sure if `priority` should be added as a computed field in the new scheme or not.

if it's not in the public API, we'll need to wait until it's declared stable (and documented) before we implement it here.

are you able to run `make docs` for this locally? the automatic generation of documentation will need to be kicked since we've updated the fields.

File: internal/provider/schema_cloudflare_email_routing_catch_all.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			Optional:    true,
+		},
+		"matcher": {
+			Description: "Matching patterns to forward to your actions.",
+			Type:        schema.TypeSet,
+			Required:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"type": {
+						Description:  "Type of matcher.",
```
Comment (jacobbednarz) [inline internal/provider/schema_cloudflare_email_routing_catch_all.go:37]:
```suggestion
						Description:  fmt.Sprintf("Type of matcher. %s", renderAvailableDocumentationValuesStringSlice([]string{"all"})),
```

(don't forget to update your imports for `fmt`)

File: internal/provider/schema_cloudflare_email_routing_catch_all.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			},
+		},
+		"action": {
+			Description: "List actions patterns.",
+			Type:        schema.TypeSet,
+			Required:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"type": {
+						Description:  "Type of supported action.",
```
Comment (jacobbednarz) [inline internal/provider/schema_cloudflare_email_routing_catch_all.go:52]:
```suggestion
						Description:  fmt.Sprintf("Type of supported action. %s", renderAvailableDocumentationValuesStringSlice([]string{"drop", "forward", "worker"})),
```

File: internal/provider/schema_cloudflare_email_routing_catch_all.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+						Type:         schema.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice([]string{"drop", "forward", "worker"}, true),
+					},
+					"value": {
+						Description: "An array with items in the following form.",
```
Comment (jacobbednarz) [inline internal/provider/schema_cloudflare_email_routing_catch_all.go:58]:
```suggestion
						Description: "A list with items in the following form.",
```

**Comment by yan12125** [comment]
Thanks! Descriptions are improved.

> if it's not in the public API, we'll need to wait until it's declared stable (and documented) before we implement it here.

Got it! Then I will omit the priority in codes for now.

> are you able to run `make docs` for this locally? the automatic generation of documentation will need to be kicked since we've updated the fields.

`docs/resources/email_routing_catch_all.md` is updated and committed. However, during `make docs`, `docs/data-sources/devices.md` is also modified, and changes in https://github.com/cloudflare/terraform-provider-cloudflare/commit/924f1e8101c6134fdb467ae8df8eac43369ac8a8#diff-410ef65dd8c8160d34c379f6a0d303020f23b6631bdd91955b2d27962e7025e9 are reverted. I assume that's unrelaed to this PR, so I didn't commit it.

**Comment by jacobbednarz** [comment]
thanks @yan12125 💯 you rock!

**Comment by yan12125** [comment]
Cool, thanks!

**Pull request merged**